chatbot-core/backend/app/routers/v1/chat.py

```diff
@@ -0,0 +1,214 @@
+from collections.abc import Generator
+
+from fastapi import APIRouter
+from fastapi import Depends
+from fastapi import HTTPException
+from fastapi import status
+from fastapi.responses import StreamingResponse
+from sqlalchemy.orm import Session
+
+from app.databases.mssql import get_db_session
+from app.models import User
+from app.models.chat import ChatMessageRequest
+from app.models.chat import ChatMessageRequestType
+from app.models.chat import ChatSessionRequest
+from app.models.chat import ChatSessionResponse
+from app.services.chat import ChatService
+from app.settings import Constants
+from app.utils.api.api_response import APIResponse
+from app.utils.api.api_response import BackendAPIResponse
+from app.utils.api.error_handler import ErrorCodesMappingNumber
+from app.utils.api.helpers import get_logger
+from app.utils.user.authentication import get_current_user
+
+
+logger = get_logger(__name__)
+router = APIRouter(prefix="/chat", tags=["chat", "session", "message"])
+
+
+@router.get("/chat-sessions", response_model=APIResponse, status_code=status.HTTP_200_OK)
+def get_chat_sessions(
+    db_session: Session = Depends(get_db_session), user: User | None = Depends(get_current_user)
+) -> None:
+    """
+    Get all chat sessions of the user.
+
+    Args:
+        db_session (Session): Database session. Defaults to relational database session.
+        user (User | None): User object
+    """
+    user_id = user.id if user else None
+
+    # Get chat sessions of user
+    chat_sessions, err = ChatService(db_session=db_session).get_chat_sessions(user_id=user_id)
+    if err:
+        status_code, detail = err.kind
+        raise HTTPException(status_code=status_code, detail=detail)
+
+    # Parse chat sessions
+    if chat_sessions:
+        data = [ChatSessionResponse.model_validate(chat_session) for chat_session in chat_sessions]
+    else:
+        data = []
+
+    return BackendAPIResponse().set_message(message=Constants.API_SUCCESS).set_data(data=data).respond()
```

Comment:
If able, I think we could modify the `max_length` of ruff config from 120 to 100. Then maybe those return lines will be formatted and splitted into multiple lines, then it can be easier to read.

Reply:
Sure! The reason I set 120 here is that it's fit long custom prompts. 100 is okay though.